### PR TITLE
Tool changes

### PR DESCRIPTION
**Pull request by b2m**
Sugggestion for some changes, new tools and configurations:

 - Remove .idea folder
 - Update .gitignore
 - Add code coverage integration to PyTest
 - Add static code analysis tools to PyTest
 - Update developer README on usage of PyTest
 - Prepare usage of black code formatter in configuration files
 - Add list of requirements for development

As the use of development tools is quite opinionated these are only suggestions. =)

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -20,3 +20,10 @@ insert_final_newline = false
 # trailing spaces in markdown indicate word wrap
 [*.md]
 trim_trailing_whitespace = false
+
+[*.py]
+multi_line_output = 3
+include_trailing_comma = True
+force_grid_wrap = 0
+use_parentheses = True
+ensure_newline_before_comments = True
```
Comment (mikegerber) [inline .editorconfig:29]: I guess these properties are documented somewhere? At least not at https://github.com/editorconfig/editorconfig/wiki/EditorConfig-Properties. But as those values are recommended at https://black.readthedocs.io/en/stable/compatible_configs.html, I guess we'll use them.

File: README-DEV.md [hunk 2]
```diff
@@ -1,10 +1,37 @@
-Testing
--------
+# Testing
+
 Use `pytest` to run the tests in [the tests directory](qurator/dinglehopper/tests):
```
Comment (mikegerber) [inline README-DEV.md:3]: I actually like the other style 😬 However, it's not _that_ important to me, so I'll merge it as is.

File: qurator/dinglehopper/tests/test_integ_ocrd_cli.py [hunk 3]
```diff
@@ -13,6 +14,7 @@
 data_dir = os.path.join(os.path.dirname(os.path.abspath(__file__)), "data")
 
 
+@pytest.mark.skipif(sys.platform == 'win32', reason="only on unix")
```
Comment (mikegerber) [inline qurator/dinglehopper/tests/test_integ_ocrd_cli.py:17]: Does the OCR-D interface really not work on win32? @kba, do you have any insight on this?

**Review by mikegerber — APPROVED** [review]
LGTM mostly, might want to touch the win32 bit in the future again

File: qurator/dinglehopper/tests/test_integ_ocrd_cli.py (hunk 3 above)
Comment (b2m) [reply qurator/dinglehopper/tests/test_integ_ocrd_cli.py:17]: The reason is related to the `AtomicWriterPerms` which uses `chmod` features only available on unix systems.

See also OCR-D/core#636.

The change in the mentioned pull requests allows the development on windows but not the execution of the functionality =)

File: README-DEV.md (hunk 2 above)
Comment (b2m) [reply README-DEV.md:3]: Was not changing the style intentionally, sry about that (makes note to check Markdown styles before changing Markdown files ;)

File: .editorconfig (hunk 1 above)
Comment (b2m) [reply .editorconfig:29]: I was also confused at first as they are also not supported by all the editors.

But they are an extension used by other tools like [isort](https://pypi.org/p/isort/).

And as far as extending the configuration does not break the support of in other tools I also thing we should keep them.

File: qurator/dinglehopper/tests/test_integ_ocrd_cli.py (hunk 3 above)
Comment (kba) [reply qurator/dinglehopper/tests/test_integ_ocrd_cli.py:17]: > @kba, do you have any insight on this?

Theoretically, most of OCR-D/core should work on Windows, but it is largely untested and we don't recommend it. But if developers like @b2m notice these platform-dependent constructs, we'll try to fix them.

**Pull request merged**